### DISCUSSION TITLE
[Help]: Reducing Delay After Drag on Embla Carousel Auto Scroll

### DISCUSSION THREAD

**Discussion by MarkForLoop** [davidjerleke/embla-carousel#994] [Help] [answered]
I’m using the [Embla Carousel auto scroll plugin](https://www.embla-carousel.com/plugins/auto-scroll/) and here’s how I’ve implemented the key areas

```js
const OPTIONS: EmblaOptionsType = { loop: true, dragFree: true }

  const [emblaRef, emblaApi] = useEmblaCarousel(options, [
    AutoScroll({
      playOnInit: true,
      speed: 1,
      stopOnInteraction: false,
      startDelay: 0,
      stopOnFocusIn: false
    })
  ])
```

The issue I’m encountering happens when I drag the carousel while it's scrolling. After dragging, it scrolls to the intended position but then pauses for a few seconds before resuming.

Is there any way to reduce or eliminate this delay after dragging?

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [x] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [X] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### CodeSandbox example

[CodeSandbox](https://codesandbox.io/p/sandbox/embla-carousel-auto-scroll-react-forked-wvpjht?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm0phswk600063b6ixsy0eb47%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm0phswk600023b6i6iai16fa%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm0phswk600033b6icn4ygzhz%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm0phswk600053b6ibqe25udk%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm0phswk600023b6i6iai16fa%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm0phswk600013b6ipl8e9lin%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fjs%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cm0phw3yb00023b6i425yjlrm%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A19%252C%2522startColumn%2522%253A44%252C%2522endLineNumber%2522%253A19%252C%2522endColumn%2522%253A44%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fjs%252FEmblaCarousel.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cm0phswk600023b6i6iai16fa%2522%252C%2522activeTabId%2522%253A%2522cm0phw3yb00023b6i425yjlrm%2522%257D%252C%2522cm0phswk600053b6ibqe25udk%2522%253A%257B%2522id%2522%253A%2522cm0phswk600053b6ibqe25udk%2522%252C%2522activeTabId%2522%253A%2522cm0phsxtp000p3b6iu6dglxrm%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm0phswk600043b6i2eaiu8uo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522cm0phsxtp000p3b6iu6dglxrm%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522cm0phswk600033b6icn4ygzhz%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cm0phswk600033b6icn4ygzhz%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by davidjerleke** [comment]
@MarkForLoop please read what the `settle` event says about `dragFree` [here](https://www.embla-carousel.com/api/events/#settle). This is what's causing the pause you mention because you've already set `startDelay` to `0`. You can manually emit the settle event like so:

```tsx
emblaApi.emit('settle');
```

...and fool the `AutoScroll` plugin that it should start before the carousel has actually settled but I would advice against it. And I haven't tested if it will cause bugs.

**Reply by MarkForLoop** [reply]
Sorry english isn't the best at times.

so I comment out the 
`startDelay` and put in `emblaApi.emit('settle');` under the useEmblaCarousel hook
is that what your saying?

Edit:
I tried what you say but it hasn't worked

**Comment by MarkForLoop** [comment]
https://github.com/user-attachments/assets/93996f64-c00d-48d1-b372-3cf9bf99dfa2

**Reply by davidjerleke** [reply, marked as answer]
@MarkForLoop you should run your code on `pointerUp`. See [this CodeSandbox](https://codesandbox.io/p/sandbox/embla-carousel-auto-scroll-react-forked-2244fq).

**Reply by MarkForLoop** [reply]
Thank you @davidjerleke
